common/beerocks/scripts/prplmesh_utils.sh.in

```diff
@@ -130,17 +134,49 @@ stop_function() {
     echo "$0: stop"
     [ `id -u` -ne 0 ] && echo "$0: warning - this commands needs root privileges so might not work (are you root?)"
 
-    [ "@BWL_TYPE@" = "DUMMY" ] && [ "$PLATFORM_INIT" == "true" ] && platform_deinit_dummy
+    [ "@BWL_TYPE@" = "DUMMY" -a "$PLATFORM_INIT" = "true" ] && platform_deinit_dummy
     [ "@BTL_TYPE@" = "LOCAL_BUS" ] && prplmesh_framework_deinit
     [ "$PRPLMESH_MODE" = "CA" -o "$PRPLMESH_MODE" = "C" ] && prplmesh_controller_stop
     [ "$PRPLMESH_MODE" = "CA" -o "$PRPLMESH_MODE" = "A" ] && prplmesh_agent_stop
 }
 
+main_agent_operational() {
+    grep -q 'CONNECTED --> OPERATIONAL' "$1/beerocks_agent.log"
+}
+
+radio_agent_operational() {
+    grep -q 'CSA' "$1/beerocks_agent_$2.log"
+}
+
+report() {
+    msg="$1"; shift
+    if "$@"; then
+        success " OK  $msg"
+    else
+        err "FAIL $msg"
+        error=true
+    fi
+}
+
 status_function() {
     echo "$0: status"
     ps -aux | grep beerocks | grep -v grep
     ps -aux | grep ieee1905_transport | grep -v grep
     ps -aux | grep local_bus | grep -v grep
+
+    # check for operational status
+    LOGS_PATH=@BEEROCKS_TMP_PATH@/logs/
+
+    report "Main agent operational" main_agent_operational $LOGS_PATH
+    [ "$error" = "true" ] && cat $LOGS_PATH/beerocks_agent.log
+
+    report "wlan0 radio agent operational" radio_agent_operational $LOGS_PATH wlan0
+    [ "$error" = "true" ] && cat $LOGS_PATH/beerocks_agent_wlan0.log
```

Comment:
This will dump the wlan0 log if the first check failed... Which maybe you want, but it is a bit weird. :ok:

Reply:
no it isn't, thanks for that